Update contribute page (#2041)
TykTechnologies/tyk-docs: CONTRIBUTING-TECHNICAL-GUIDE.md

* Fetch from the remote branch `git fetch upstream`
* Rebase your branch with the latest remote branch content `git rebase upstream/master`

The following guide briefly explains how to work with Hugo, you would then need push to your forked repository and then create a Pull Request to Tyk's `master` branch.

## How to Use

Our Documentation is constructed using [Hugo](http://gohugo.io/).

## Getting Started
1. Clone this repository 
2. Navigate to the project directory

### Use with Docker
1. [Docker](https://docs.docker.com/get-docker/)
2. Run `docker-compose up` from the project directory

### Use with Hugo
1. [Install Hugo v0.96.0 or greater](https://github.com/gohugoio/hugo/releases)
2. Run `hugo server --theme=tykio --buildDrafts --enableGitInfo` from the `tyk-docs/tyk-docs` directory

Go to  [http://localhost:1313/docs/nightly/](http://localhost:1313/docs/nightly/) to view the docs locally

## Adding and Editing Content

The docs content lives in `tyk-docs/content`.

### Add a new Section

1. Add a new folder in within the `tyk-docs/tyk-docs/content/` Directory. For example `new-section`

> 2. Within your new folder create a markdown file with the following command from your terminal - `hugo new` . So for the above, use `hugo new /new-section/new-section.md`. This file will be converted to the equivalent of an `index.html` file.

Author: ```suggestion
2. Within your new folder create a markdown file with the following command from your terminal - `hugo new`. For the above example you would run `hugo new /new-section/new-section.md`. This file will be converted to the equivalent of an `index.html` file.
```

* `description` is used for the meta description in the HTML output
* `menu` is used to place the page in the correct place within the navigation hierarchy. By default a new page is assigned to the root level (`main`)
* `weight` is used to order pages within a section of the menu with `0` being the top level page within a section.

You can create a dynamic, nested navigation hierarchy simply by changing the `parent` field to the name of the parent page. Note, **these names must be unique**.

Example front matter for a page:

```
---
title: "Test"
date: 2021-02-10
tags: ["Tyk", "advanced-configuration", "Dashboard"]
description: "Testing the description and tagging functionality in Tyk"
menu:
  main:
    parent: "Advanced Configuration"
weight: 10
---
```

## Content

The content itself is just markdown that follows the front matter block. When you add and edit new content, Hugo should auto-reload and you should be able to see the changes live in your browser (if not, refresh). Sometimes Hugo gets confused and you may need to re-run it.

## Shortcodes

> There are various shortcodes used within the Tyk documentation.

Author: ```suggestion
Various shortcodes are used within the Tyk documentation.
```
